libtransmission/CMakeLists.txt

```diff
@@ -217,6 +214,19 @@ tr_allow_compile_if(
         file-posix.cc
         subprocess-posix.cc)
 
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Comment:
`tr_allow_compile_if()` exists to avoid writing `if()`s. Also, using `TRUE` condition makes little sense as the value is constant (and so the call makes no difference); I know I used it myself for crypto-utils-fallback.cc but nobody's perfect. I'd say it should be

```patch
@@ -204,10 +204,10 @@
         watchdir-inotify.cc
     [=[[WITH_KQUEUE]]=]
         watchdir-kqueue.cc
-    [=[[APPLE]]=]
+    [=[[APPLE AND CMAKE_CXX_COMPILER_ID MATCHES "Clang"]]=]
         tr-assert.mm
         utils.mm
-    [=[[NOT APPLE]]=]
+    [=[[NOT (APPLE AND CMAKE_CXX_COMPILER_ID MATCHES "Clang")]]=]
         tr-assert.cc
     [=[[WIN32]]=]
         file-win32.cc
```

or maybe

```patch
@@ -187,6 +187,11 @@
     PRIVATE
         ${CMAKE_CURRENT_BINARY_DIR}/version.h)
 
+set(APPLE_CLANG OFF)
+if(APPLE AND CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    set(APPLE_CLANG ON)
+endif()
+
 tr_allow_compile_if(
     [=[[TRUE]]=]
         crypto-utils-fallback.cc
@@ -204,10 +209,10 @@
         watchdir-inotify.cc
     [=[[WITH_KQUEUE]]=]
         watchdir-kqueue.cc
-    [=[[APPLE]]=]
+    [=[[APPLE_CLANG]]=]
         tr-assert.mm
         utils.mm
-    [=[[NOT APPLE]]=]
+    [=[[NOT APPLE_CLANG]]=]
         tr-assert.cc
     [=[[WIN32]]=]
         file-win32.cc
```

Then again, I'm not convinced that tr-assert.mm needs to be disabled, there's no ARC used there and GCC should compile it just fine. The program is also expected to terminate as a result of `tr_assert_report()` call, so it doesn't matter much if there is a leak or not because of lack of ARC support. That is to say, I think I'd rather have it as

```patch
@@ -206,9 +206,10 @@
         watchdir-kqueue.cc
     [=[[APPLE]]=]
         tr-assert.mm
-        utils.mm
     [=[[NOT APPLE]]=]
         tr-assert.cc
+    [=[[APPLE AND CMAKE_CXX_COMPILER_ID MATCHES "Clang"]]=]
+        utils.mm
     [=[[WIN32]]=]
         file-win32.cc
         subprocess-win32.cc
```

Reply:
@mikedld On the other hand, what’s wrong with Unix file, why at all `tr-assert.mm`, especially in non-Cocoa build? (And for w/e it does not build, at least at the moment, triggering `libfmt` errors.)

Reply:
> @mikedld On the other hand, what’s wrong with Unix file, why at all `tr-assert.mm`, especially in non-Cocoa build? (And for w/e it does not build, at least at the moment, triggering `libfmt` errors.)

If you know any other way to call releated macOS APIs - feel free to share with us.
We need to raise exception with readable crash report.

Reply:
done in 6deee3f1c080718c24b0253b94c7b91a8bfe93fd

And I prefer to keep utils.mm for clang. Let's recall that barracuda is trying to build on macOS 10.6, that is, an OS released in 2009 and for which we don't provide support, so we won't be looking at the crash reports for it.